src/Features/CSharp/Portable/MakeStructFieldsWritable/CSharpMakeStructFieldsWritableDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,118 @@
+
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.CodeQuality;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeAnalysis.CSharp.MakeStructFieldsWritable
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed class CSharpMakeStructFieldsWritableDiagnosticAnalyzer : AbstractCodeQualityDiagnosticAnalyzer
+    {
+        private static readonly DiagnosticDescriptor s_diagnosticDescriptor = CreateDescriptor(
+            IDEDiagnosticIds.MakeStructFieldsWritable,
+            new LocalizableResourceString(nameof(FeaturesResources.Make_readonly_fields_writable), FeaturesResources.ResourceManager, typeof(FeaturesResources)),
+            new LocalizableResourceString(nameof(FeaturesResources.Make_readonly_fields_writable), FeaturesResources.ResourceManager, typeof(FeaturesResources)),
+            isUnneccessary: false);
+
+        public CSharpMakeStructFieldsWritableDiagnosticAnalyzer()
+            : base(ImmutableArray.Create(s_diagnosticDescriptor), GeneratedCodeAnalysisFlags.ReportDiagnostics)
+        {
+        }
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+            => DiagnosticAnalyzerCategory.SemanticDocumentAnalysis;
+
+        public override bool OpenFileOnly(Workspace workspace) => true;
+
+        protected override void InitializeWorker(AnalysisContext context)
+        {
+            context.RegisterCompilationStartAction(compilationStartContext
+                => SymbolAnalyzer.CreateAndRegisterActions(compilationStartContext));
+        }
+
+        private sealed class SymbolAnalyzer
+        {
+            private bool _hasTypeInstanceAssigment;
+            private readonly INamedTypeSymbol _namedTypeSymbol;
+            private SymbolAnalyzer(INamedTypeSymbol namedTypeSymbol)
+            {
+                _namedTypeSymbol = namedTypeSymbol;
+            }
+
+            public static void CreateAndRegisterActions(CompilationStartAnalysisContext compilationStartContext)
+            {
+                compilationStartContext.RegisterSymbolStartAction(symbolStartContext =>
+                {
+                    // We report diagnostic only if these requirements are met:
+                    // 1. The type is struct
+                    // 2. Struct contains at least one 'readonly' field
+                    // 3. Struct contains assignment to 'this' outside the scope of constructor
+                    var namedTypeSymbol = (INamedTypeSymbol)symbolStartContext.Symbol;
+
+                    // We are only interested in struct declarations
+                    if (namedTypeSymbol.TypeKind != TypeKind.Struct)
+                    {
+                        return;
+                    }
+                    //We check if struct contains any 'readonly' fields
```

Comment:
blank line above this.

Reply:
Resolved in [https://github.com/dotnet/roslyn/pull/32063/commits/26fbaad4abed1f8e1c7b663a08a91696cebfe036](https://github.com/dotnet/roslyn/pull/32063/commits/26fbaad4abed1f8e1c7b663a08a91696cebfe036)